From e375930813f0b8663ec4ceb6d5ea1850fbbe9695 Mon Sep 17 00:00:00 2001
From: Hui Wang <hui.wang@canonical.com>
Date: Mon, 31 May 2021 13:08:19 +0200
Subject: [PATCH] alsa-mixer: only use switch to mute Front in the Headphone
 path

According to the alsa-info.txt in the pipewire issues of #747 and #1206,
the Front Playback Volume is shared by Headphone and Lineout or
Headphone and Speaker, But Headphone, Lineout or Speaker they all have
independent Playback Switch, change to only use switch to mute the
Lineout or Speaker. This could resolve the issues of #747 and #1206.

See #1206 and #747
---
 spa/plugins/alsa/mixer/paths/analog-output-headphones-2.conf | 5 ++++-
 spa/plugins/alsa/mixer/paths/analog-output-headphones.conf   | 5 ++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/spa/plugins/alsa/mixer/paths/analog-output-headphones-2.conf b/spa/plugins/alsa/mixer/paths/analog-output-headphones-2.conf
index e98fe2537..bda137d3c 100644
--- a/spa/plugins/alsa/mixer/paths/analog-output-headphones-2.conf
+++ b/spa/plugins/alsa/mixer/paths/analog-output-headphones-2.conf
@@ -84,9 +84,12 @@ volume = off
 switch = off
 volume = off
 
+; On some machines, the Front Volume Control is shared by Headphone and Lineout,
+; or Headphone and Speaker, but they have independent Volume Switch. Here only
+; use switch to mute Lineout or Speaker.
 [Element Front]
 switch = off
-volume = off
+volume = zero
 
 [Element Rear]
 switch = off
diff --git a/spa/plugins/alsa/mixer/paths/analog-output-headphones.conf b/spa/plugins/alsa/mixer/paths/analog-output-headphones.conf
index ac4ec659a..3c62c5e67 100644
--- a/spa/plugins/alsa/mixer/paths/analog-output-headphones.conf
+++ b/spa/plugins/alsa/mixer/paths/analog-output-headphones.conf
@@ -126,9 +126,12 @@ volume = off
 switch = off
 volume = off
 
+; On some machines, the Front Volume Control is shared by Headphone and Lineout,
+; or Headphone and Speaker, but they have independent Volume Switch. Here only
+; use switch to mute Lineout or Speaker.
 [Element Front]
 switch = off
-volume = off
+volume = zero
 
 [Element Rear]
 switch = off
-- 
GitLab

